### PR TITLE
Including code review suggestions.

### PR DESCRIPTION
**Pull request by chamindias**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 1]
```diff
@@ -2432,43 +2436,76 @@ private boolean addExternalAPIStoresDetails(APIIdentifier apiId,Set<APIStore> ap
...
 
-            for (String gatewayEnvironmentName : gatewyEnvironments.keySet()) {
-                Environment currentGatewayEnvironment = gatewyEnvironments.get(gatewayEnvironmentName);
+            if (clusteringAgent != null) {
+                //changing stat publishing status at other nodes via a cluster message
+                try {
+                    clusteringAgent.sendMessage(new StatUpdateClusterMessage(updatedStatus), true);
+                } catch (ClusteringFault clusteringFault) {
+                    //error is only logged because initially gateway has modified the status
+                    log.error("Failed to send cluster message to Publisher/Store domain " +
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: log with exception

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 2]
```diff
@@ -2432,43 +2436,76 @@ private boolean addExternalAPIStoresDetails(APIIdentifier apiId,Set<APIStore> ap
...
+                try {
+                    //get the stub and the call the admin service with the credentials
+                    GatewayStatsUpdateServiceStub stub =
+                            new GatewayStatsUpdateServiceStub(gatewayServiceUrl + "GatewayStatsUpdateService");
+                    ServiceClient gatewayServiceClient = stub._getServiceClient();
+                    CarbonUtils.setBasicAccessSecurityHeaders(gatewayUserName, gatewayPassword, gatewayServiceClient);
+                    stub.updateStatPublishGateway(receiverUrl, user, password, updatedStatus);
+                } catch (AxisFault e) {
+                    //error is only logged because the process should be executed in all gateway environments
+                    log.error("Error in calling Stats update web service in Gateway Environment." + e.getMessage());
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: log with exception

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 3]
```diff
@@ -2432,43 +2436,76 @@ private boolean addExternalAPIStoresDetails(APIIdentifier apiId,Set<APIStore> ap
...
-                CarbonUtils.setBasicAccessSecurityHeaders(gatewayUserName, gatewayPassword, gatewayServiceClient);
-
-                //send an empty string if at least one mandatory parameter is null
-                if (receiverUrl == null || user == null || password == null) {
-                    receiverUrl = "";
-                    user = "";
-                    password = "";
+                try {
+                    //get the stub and the call the admin service with the credentials
+                    GatewayStatsUpdateServiceStub stub =
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: better move service name to a onstant

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 1 above)
Comment (chamindias) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Fixed in : https://github.com/chamindias/carbon-apimgt/commit/2917bb535a4686418310b787af361524ff352252

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 2 above)
Comment (chamindias) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Fixed in : https://github.com/wso2/carbon-apimgt/commit/67c064f7410655acdbfedac15735038bc54ddaa7

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 3 above)
Comment (chamindias) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Fixed in : https://github.com/wso2/carbon-apimgt/commit/67c064f7410655acdbfedac15735038bc54ddaa7

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 4]
```diff
@@ -2432,43 +2436,76 @@ private boolean addExternalAPIStoresDetails(APIIdentifier apiId,Set<APIStore> ap
...
+                            new GatewayStatsUpdateServiceStub(gatewayServiceUrl + "GatewayStatsUpdateService");
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Please add a constant.

Comment (chamindias) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Fixed in : https://github.com/wso2/carbon-apimgt/commit/67c064f7410655acdbfedac15735038bc54ddaa7

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 2 above)
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: The Log message should indicate which Gateway update Failed.

Comment (chamindias) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Done in : https://github.com/wso2/carbon-apimgt/pull/1382

**Pull request merged**